totalRP3_Extended_Tools/script/effects/effects.lua

```diff
@@ -740,6 +740,181 @@ local function speech_player_init()
+	local currentCharacter;
+	for _, emote in ipairs(input) do
+		local characterList = {}
+		for character in string.gmatch(emote[1], "([%z\1-\127\194-\244][\128-\191]*)") do
+			table.insert(characterList, Utils.str.convertSpecialChars(character));
+		end
+
+		if not currentCharacter or currentCharacter ~= characterList[2] then
+			currentCharacter = characterList[2];
+			table.insert(characterTableSorted, { characterList[2], { emote }});
+		else
+			table.insert(characterTableSorted[#characterTableSorted][2],  emote);
+		end
+	end
+
+	local currentCount = 0;
+	local currentTable;
+	local firstCharacter;
+	local previousCharacter;
+	for _, characterTable in ipairs(characterTableSorted) do
+		local character = characterTable[1];
+		local emotes = characterTable[2];
+		local characterSize = #emotes;
+		if (currentCount + characterSize) > maxSize then
+			-- Adding the previous table
+			if currentTable then
+				if firstCharacter ~= previousCharacter then
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter) .. "-" .. string.upper(previousCharacter), currentTable });
+				else
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter), currentTable });
+				end
+			end
+
+			if characterSize > maxSize then
+				-- Split into smaller tables
+				firstCharacter = nil;
+				local i = 0;
+				while (maxSize * i) < characterSize do
+					i = i + 1;
+					currentTable = {};
+					local offset = (i - 1) * maxSize;
+					for j = 1, min(maxSize, characterSize - offset) do
+						table.insert(currentTable, emotes[offset + j]);
+					end
+					table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(character) .. i, currentTable });
+				end
+				currentTable = nil;
+				currentCount = 0;
+			else
+				currentTable = {};
+				for i = 1, characterSize do
+					table.insert(currentTable, emotes[i]);
+				end
+				firstCharacter = character;
+				currentCount = characterSize;
+			end
+			previousCharacter = character;
+		else
+			if not currentTable then
+				currentTable = {};
+			end
+			for i = 1, characterSize do
+				table.insert(currentTable, emotes[i]);
+			end
+			currentCount = currentCount + characterSize;
+
+			if not firstCharacter then
+				firstCharacter = character;
+			end
+			previousCharacter = character;
+		end
+	end
+
+	-- Adding the previous table
+	if currentTable then
+		if firstCharacter ~= previousCharacter then
+			table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter) .. "-" .. string.upper(previousCharacter), currentTable });
+		else
+			table.insert(output, { loc.EFFECT_DO_EMOTE_OTHER .. " " .. string.upper(firstCharacter), currentTable });
+		end
+	end
+
+	return output;
+end
+
+local function do_emote_init()
+	local editor = TRP3_EffectEditorDoEmote;
+
+	-- Build list of emotes
+	local spokenEmotes = {}
+	for _, emoteToken in ipairs(TextEmoteSpeechList) do
+		spokenEmotes[emoteToken] = true;
+	end
```

Comment:
If the value assigned to the key inside the `spokenEmotes` table isn't important, you could simplify this to:
```suggestion
	local spokenEmotes = tInvert(TextEmoteSpeechList);
```